src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpCodeVisitor.cs

```diff
@@ -16,6 +15,7 @@ public class CSharpCodeVisitor : CodeVisitor<CSharpCodeWriter>
         private const string ItemParameterName = "item";
         private const string ValueWriterName = "__razor_attribute_value_writer";
         private const string TemplateWriterName = "__razor_template_writer";
+        private static readonly int MaxStringLiteralLength = 499; // Max 1k string size despite x86/x64
```

Comment:
Decided on 1k. Definitely open to other suggestions, this in all reality is just one of the many magic numbers that fit in the small object heap.